Formula/semgrep.rb

```diff
@@ -30,7 +30,6 @@ class Semgrep < Formula
   depends_on "pkg-config" => :build
   depends_on "pcre"
   depends_on "python@3.9"
-  depends_on "tree-sitter"
```

Comment:
I think you still need this.

Reply:
We tree-sitter version 0.19. Can we just specify:
```
depends_on "tree-sitter@0.19"
```
?

Reply:
We don't ship tree-sitter 0.19 because that version isn't maintained.

Can we regenerate your parsers using our tree-sitter instead?

Reply:
Not easily: Generating the parsers is a complex and brittle process, so we generate them ahead of time, making them dependent on a specific version of the tree-sitter API.

I'm pleasantly surprised that our parsers generated for tree-sitter 0.19 work with tree-sitter 0.20 as provided by homebrew. I think this is how it's supposed to work, but months ago I observed compatibility issues across earlier versions.

For now, let's keep using `depends_on "tree-sitter"` which is going to be version 0.20.0 or later.

Reply:
Great. Let's keep this setup for now. If there are user complaints about things being broken, we can take steps to address those then.

I'm hoping @brendongo's new test will help us catch API incompatibility before any users even see it too. (It will be run any time there are changes to the `tree-sitter` formula.)